src/firebase-db.ts

```diff
@@ -74,7 +110,85 @@ const watchCollection = (path: string, portalData: IPortalData, listener: IDocum
   });
 };
 
-export const watchAnswers = (portalData: IPortalData, listener: IDocumentsListener) => {
-  const answersPath = `sources/${portalData.database.sourceKey}/answers`;
-  watchCollection(answersPath, portalData, listener);
+export const addDBListener = (path: string, listener: DBChangeListener) => {
+  if (!listeners[path]) {
+    listeners[path] = [];
+  }
+  listeners[path].push(listener);
+
+  // notify right away if we already have data (asynchronously, so that a listener that wants
+  // to immediately unsubscribe will get the unsubscribe method first).
+  if (localDB[path]) {
+    setTimeout(() => {
+      listener(localDB[path]);
+    }, 0);
+  }
+
+  const unsubscribe = () => {
+    const idx = listeners[path].indexOf(listener);
+    if (idx > -1) {
+      listeners[path].splice(idx, 1);
+    }
+  };
+  return unsubscribe;
+};
+
+const notifyListeners = (path: string, value: any) => {
+  listeners[path]?.forEach(listener => listener(value));
+};
+
+// returns value immediately if it exists, or gets the value once on the first update.
+// equivalent to `firebase.once`
+export const getCurrentDBValue = (path: string, listener: DBChangeListener) => {
```

Comment:
This would probably be better if it returned a promise that resolved to the value.  Then someone who just wants to get the current value doesn't need to create a DBChangeListener. Just a suggestion though, no need to change.